src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +113,828 @@ public static void init()
 
     /**
      * Loads all styles saved in ["/assets/minecolonies/schematics/"].
-     * Puts these in {@link #hutStyleMap}, with key being the name of the hutDec (E.G. Lumberjack).
-     * and the value is a list of styles. Puts decorations in {@link #decorationStyleMap}.
+     * Puts these in {@link #md5Map}, with key being the fullname of the structure (schematics/stone/Builder1).
      */
     private static void loadStyleMaps()
     {
-        try
+        if (!Configurations.ignoreSchematicsFromJar)
         {
-            @NotNull final URI uri = ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI();
-            final Path basePath;
-
-            if ("jar".equals(uri.getScheme()))
+            try (FileSystem fileSystem = FileSystems.newFileSystem(ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI(), Collections.emptyMap()))
             {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap()))
-                {
-                    basePath = fileSystem.getPath(SCHEMATICS_ASSET_PATH);
-                    loadStyleMaps(basePath);
-                }
+                final Path basePath = fileSystem.getPath(SCHEMATICS_ASSET_PATH);
+                loadSchematicsForPrefix(basePath, SCHEMATICS_PREFIX);
             }
-            else
+            catch (IOException | URISyntaxException e)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)